### ISSUE TITLE
Crash Proxy using Ruby

### ISSUE THREAD

**Issue by bsharpe** [rethinkdb/rethinkdb#5904]
Working on a small reproducible case for this, but the general gist is:

- connect to a proxy that's connected to a cluster 
- using the ruby driver, create a cursor on a table with a large number of row (not sure that's relevant)
- save this data to a different connection to a different node (copying data from remote to local)
- Ctrl-C cancel the import process
- remote Proxy dies every time
- tend to think it's the Ruby driver not getting a chance to close the connection properly

```
2016-06-27T14:39:05.864845985 317.677548s error: Error in src/errors.cc at line 161:
2016-06-27T14:39:05.864871941 317.677574s error: Uncaught exception of type "tcp_conn_read_closed_exc_t"\n  what(): Network connection read end closed
2016-06-27T14:39:05.864879717 317.677582s error: Backtrace:
2016-06-27T14:39:05.974798536 317.787517s error: Mon Jun 27 14:39:05 2016\n\n1 [0xabb110]: backtrace_t::backtrace_t() at ??:?\n2 [0xabb4a3]: format_backtrace(bool) at ??:?\n3 [0xd3
2016-06-27T14:39:05.974866901 317.787569s error: Exiting.
```

**Comment by danielmewes** [comment]
Thanks for the bug report @bsharpe .

This is probably related to the connection being dropped at an unexpected moment. However it clearly shouldn't crash the server.

Which version of RethinkDB are you running? Which operating system / distribution is this on? Is the Ruby client connecting to localhost or over a network?

**Comment by bsharpe** [comment]
RDBv 2.3.4
Server -> Linux db-1 4.2.0-38-generic #45~14.04.1-Ubuntu SMP Thu Jun 9 09:27:51 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
Client -> OS X, Ruby 2.3.1

Connection is over a network.

Have not been able to reproduce it locally.

**Comment by danielmewes** [comment]
A likely fix for this is now in internal code review 3685.
It will ship with RethinkDB 2.3.5.

The issue appears to be specific to TLS-encrypted connections.

**Comment by danielmewes** [comment]
Fixed in branch next via 4cc466f and v2.3.x via 04fa66b.

**Comment by danielmewes** [comment]
@bsharpe We've released RethinkDB 2.3.5 today with a fix for this crash.
